aikiblog/views.py

```diff
@@ -213,3 +215,58 @@ def add_techniques(request):
         return HttpResponseRedirect('/add_techniques#content')
     else:
         return render(request, 'add_techniques.html', {'form': form, 'techniques': techniques})
+
+
+def calendar(request, year=None):
+    if year: year = int(year)
+    else:    year = time.localtime()[0]
+
+    nowy, nowm = time.localtime()[:2]
+    lst = []
+
+    for y in [year-2, year-1, year]:
+        mlst = []
+        for n, month in enumerate(mnames):
+            entry = current = False   # are there entry(s) for this month; current month?
+            entries = Training.objects.filter(date__year=y, date__month=n+1, user=request.user)
+
+            if entries:
+                entry = True
+            if y == nowy and n+1 == nowm:
+                current = True
+            mlst.append(dict(n=n+1, name=month, entry=entry, current=current, entries=entries))
+        lst.append((y, mlst))
+
+    return render(request, "calendar.html", {'years': lst, 'user': request.user, 'year': year})
+
+
+def month(request, year, month, change=None):
+    year, month = int(year), int(month)
+
+    if change in ("next", "prev"):
+        now, mdelta = date(year, month, 15), timedelta(days=31)
+        if change == "next":   mod = mdelta
+        elif change == "prev": mod = -mdelta
+
+        year, month = (now+mod).timetuple()[:2]
+
+    cal = calendar_library.Calendar()
+    month_days = cal.itermonthdays(year, month)
+    nyear, nmonth, nday = time.localtime()[:3]
+    lst = [[]]
+    week = 0
+
+    for day in month_days:
+        trainings = current = False
+        if day:
+            trainings = Training.objects.filter(date__year=year, date__month=month, date__day=day, user=request.user)
+            if day == nday and year == nyear and month == nmonth:
+                current = True
+
+        lst[week].append((day, trainings, current))
+        if len(lst[week]) == 7:
+            lst.append([])
+            week += 1
+
+    return render(request, "month.html", {'year': year, 'month': month, 'user': request.user,
+                        'month_days': lst, 'mname': mnames[month-1]})
```

Comment:
Trochę nie czaję z tym kalendarzem. Chodzi o to, że tworzysz go sam, a nie korzystasz z gotowca, czy jak?